waspc/src/Wasp/Generator/Job/Process.hs

```diff
@@ -95,20 +97,21 @@ runProcessAsJob process jobType chan =
       return $ ExitFailure 1
 
 runNodeCommandAsJob :: Path' Abs (Dir a) -> String -> [String] -> J.JobType -> J.Job
-runNodeCommandAsJob fromDir command args jobType chan = do
-  errorOrNodeVersion <- getNodeVersion
-  case errorOrNodeVersion of
+runNodeCommandAsJob = runNodeCommandAsJobWithExtraEnv []
+
+runNodeCommandAsJobWithExtraEnv :: [(String, String)] -> Path' Abs (Dir a) -> String -> [String] -> J.JobType -> J.Job
+runNodeCommandAsJobWithExtraEnv extraEnvVars fromDir command args jobType chan =
+  getNodeVersion >>= \case
     Left errorMsg -> exitWithError (ExitFailure 1) (T.pack errorMsg)
     Right nodeVersion ->
       if SV.isVersionInRange nodeVersion C.nodeVersionRange
         then do
-          let process = (P.proc command args) {P.cwd = Just $ SP.fromAbsDir fromDir}
-          runProcessAsJob process jobType chan
-        else
-          exitWithError
-            (ExitFailure 1)
-            (T.pack $ makeNodeVersionMismatchMessage nodeVersion)
+          envVars <- getAllEnvVars
+          let nodeCommandProcess = (P.proc command args) {P.env = Just envVars, P.cwd = Just $ SP.fromAbsDir fromDir}
+          runProcessAsJob nodeCommandProcess jobType chan
+        else exitWithError (ExitFailure 1) (T.pack $ makeNodeVersionMismatchMessage nodeVersion)
   where
+    getAllEnvVars = (++ extraEnvVars) <$> getEnvironment
```

Comment:
Was `getEnvironment` the old, implicit behavior before when we did not supply any `P.env` value?

Reply:
That's right.

A `Process` takes an `env` field of type `Maybe [(String, String)]`:
- When you give it `Nothing`, it acts as if you haven't even specified the `env` field and inherits the environment from the parent.
- When you give it `Just someList`, it sees _exclusively the stuff in the list_. Therefore, if we wish to extend the environment (and not override it), we need to make sure our list also contains the parent process's environment.

I noticed this because my functions were failing with a "missing `DATABASE_URL`" error.

Reply:
Makes sense, thanks! 👍🏻